### PR TITLE
Add Parameter to Exclude DLC Characters

### PR DESCRIPTION
**Pull request by 10xJosh**
This is in regards to issue #4 . I added the option to the list of parameters and added the variable needed for the option.

I wasn't sure if you wanted me to do the implementation with removing the DLC characters from the list as well so I just figured I'd send the pull request in case this is all you wanted :) If you want me to do that I can, just let me know!

### CODE REVIEW

**Comment by 10xJosh** [comment]
Made some changes to the readme but I accedeintly moved it to this branch. I reverted it so that the pull request only adds the Add parameter and not the readme. sorry for the messy commits...

File: MkBuildBot/Classes/CommandRegisterer.cs
Comment (Founntain) [inline MkBuildBot/Classes/CommandRegisterer.cs:57]: Fix indentation here

File: MkBuildBot/Program.cs
Comment (Founntain) [inline MkBuildBot/Program.cs:80]: Same as above and why are here two blank lines?

As I don't have anything against using the specific type for the variable I would like to use `var` here as well, as it fits the code style here. Not forgetting to mention that the order is wrong. `excludeInline` should be before `isDLCCharactersExcluded` as it should match the order as above, same goes for `amount`

Also not blank spaces here as they all belong together.

**Review by Founntain — CHANGES_REQUESTED** [review]
> *I wasn't sure if you wanted me to do the implementation with removing the DLC characters from the list as well*

Well the implementation of #4 should obviously contain the actual implementation of said exclusion, it is pretty senseless to just add the parameter and doing absolutely nothing :) Also before you continue fix your merge conflicts

Also fix codefactor issues

**Comment by 10xJosh** [comment]
There is no formating issue in the code, github is just being silly

2023-11-22 08_55_31-MarioKart8DX-BuildGenerator - Microsoft Visual Studio

**Comment by 10xJosh** [comment]
> Well the implementation of #4 should obviously contain the actual implementation of said exclusion, it is pretty senseless to just add the parameter and doing absolutely nothing :) Also before you continue fix your merge conflicts

I'm very sorry I misunderstood what you meant. You mentioned that this would be a tiny tiny change, so I thought you were already working on the implementation of excluding the DLC characters. I looked at the code and the characters are pulled randomly by .png file. 

To fully implement the change requires changing the main functionality of how the characters are selected. Passing the DLC characters variable through multiple function parameters and then comparing whatever .png file is selected to a list of DLC characters. 

With the amount of functions that need to be altered, It didnt seem like it was a tiny change but a big one, so I opted to just post this. Very sorry if I wasted your time with this.

**Comment by Founntain** [comment]
> There is no formating issue in the code, github is just being silly
> 
> 2023-11-22 08_55_31-MarioKart8DX-BuildGenerator - Microsoft Visual Studio
> 

Ill check in a couple of minutes if it is github beeing silly. However this shouldn't be happening in GitHub.

**Comment by Founntain** [comment]
In the end this is a very tiny issue as it is just an extra check. We have examples how inline bikes is included, its exactly the same as that just with those extra characters in a extra method.

The reason why I said tiny is, because it is not much. Just the extra command and those extra chars that have to be excluded.

File: MkBuildBot/Program.cs
Comment (Founntain) [inline MkBuildBot/Program.cs:84]: I do get why you did the bool check in this line.

However I would love to seperate these here as well. (Like it was done with excludeInlineParam) this gives us the ability if we actually need the param value for whatever reason.

**Review by Founntain — CHANGES_REQUESTED** [review]
In the end I can't approve this changes yet, not because of the requested change, more because the intended implementation is not yet implemented and it makes no sense to me to merge something that is supposed to implement the logic as said in #4, that the exclusion logic needs to be implemented inside the generation.

So it is up to you to finish it otherwise I can't and won't merge it. I want to be honest as well if you don't finish it properly as intended this seemed more than a waste of time for you than me.

**Comment by Founntain** [comment]
> There is no formating issue in the code, github is just being silly

Well I did a check now and it was indeed wrong intended, so it's not GitHub beeing silly. My IDE showed it correctly however when opening the file raw it was showed broken. It might be your git that messed it up as this is sometimes a common cause, so check that your IDE uses **UTF-8** Encoding and **CRLF** Line Endings

**Comment by 10xJosh** [comment]
> > There is no formating issue in the code, github is just being silly
> 
> Well I did a check now and it was indeed wrong intended, so it's not GitHub beeing silly. My IDE showed it correctly however when opening the file raw it was showed broken. It might be your git that messed it up as this is sometimes a common cause, so check that your IDE uses **UTF-8** Encoding and **CRLF** Line Endings

Thanks for bringing this to my attention. I think I fixed it.

**Comment by 10xJosh** [comment]
> In the end I can't approve this changes yet, not because of the requested change, more because the intended implementation is not yet implemented and it makes no sense to me to merge something that is supposed to implement the logic as said in #4, that the exclusion logic needs to be implemented inside the generation.
> 
> So it is up to you to finish it otherwise I can't and won't merge it. I want to be honest as well if you don't finish it properly as intended this seemed more than a waste of time for you than me.

I think its best that I don't continue any further with this issue for both of our sakes. I think my style of programming will only cause more issues instead of being helpful. 

With that being said, I did learn a lot from your code and appreciate you taking the time to look over my changes. This is an awesome project and I wish the best on this implementation!

**Comment by Founntain** [comment]
> > In the end I can't approve this changes yet, not because of the requested change, more because the intended implementation is not yet implemented and it makes no sense to me to merge something that is supposed to implement the logic as said in #4, that the exclusion logic needs to be implemented inside the generation.
> > So it is up to you to finish it otherwise I can't and won't merge it. I want to be honest as well if you don't finish it properly as intended this seemed more than a waste of time for you than me.
> 
> I think its best that I don't continue any further with this issue for both of our sakes. I think my style of programming will only cause more issues instead of being helpful.
> 
> With that being said, I did learn a lot from your code and appreciate you taking the time to look over my changes. This is an awesome project and I wish the best on this implementation!

Thanks and good look anyways. Maybe you should check the [official C# coding conventions and style](https://learn.microsoft.com/en-us/dotnet/csharp/fundamentals/coding-style/identifier-names) as this the style I'm using for the sake of simplicity and being open to everyone to contribute.

In the end I have nothing against new programmers helping out. However when contributing there should be an understanding of the git workflow, apply changes correctly and most important: *the projects that the person wants to contribute is a project where they have some understanding of the topic as well in this example Mario Kart 8 Deluxe*

**Pull request closed without merging**
